cmd/ldap-users-from-github-owners-files/main.go

```diff
@@ -22,30 +22,41 @@ func main() {
 	ldapFile := flag.String("ldap-file", "", "File generated via ldapsearch -LLL -x -h ldap.corp.redhat.com -b ou=users,dc=redhat,dc=com '(rhatSocialURL=GitHub*)' rhatSocialURL uid")
 	repoBaseDir := flag.String("repo-base-dir", "", "base dir for the target repo. Will be used to resolve OWNERS_ALIASES")
 	repoSubdir := flag.String("repo-sub-dir", ".", "Subdir relative to the --repo-base-dir to look for OWNERS files")
+	mappingFile := flag.String("mapping-file", "", "File to store the mapping m(github_login)=kerberos_id. When this flag is provided, the program exists after generating the mapping file.")
 	flag.Parse()
 
-	ldapUsers, errs := getLdapUsers(*ldapFile, *repoBaseDir, *repoSubdir)
+	mapping, errs := createLDAPMapping(*ldapFile)
 	if len(errs) > 0 {
 		for _, err := range errs {
-			logrus.WithError(err).Error("encountered error trying to resolve owners")
+			logrus.WithError(err).Warn("encountered error trying to parse ldap file")
 		}
 	}
-	serialized, err := json.Marshal(ldapUsers.List())
-	if err != nil {
-		logrus.WithError(err).Fatal("failed to serialize ldap user list")
+
+	if *mappingFile != "" {
```

Comment:
Do we want to add a unit test for the scenario where we're generating the mapping file?

Reply:
Updated.